internal/ai/conversation.go

```diff
@@ -18,6 +19,11 @@ type Conversation struct {
 	Messages     []conversationTurn
 
 	maxTokens int64 // Maximum number of output tokens per response
+	
+	// Token tracking for summarization
+	totalInputTokens    int64
+	totalCacheReadTokens int64
+	tokenLimit          int64 // When to trigger summarization
```

Comment:
We don't need to sum the token usage of each turn in the conversation. Each turn is cumulative; each turn we send the entire conversation history to the AI. Therefore we only need to look at the token usage in the most recent turn when determining when to summarize